### ISSUE TITLE
TriStateCheckbox: Custom Icon for undetermined state

### ISSUE THREAD

**Issue by ig-onoffice-de** [primefaces/primevue#3418]
### Describe the feature you would like to see added

In #3301 there was added the possibility to pass custom icons for true and false state of TriStateCheckbox. For the undetermined value it is not possible to pass a custom icon. 

### Is your feature request related to a problem?

It's related to usability. The normal user knows the icons as 
true ✔
false 🔳
undetermined ➖
We would like to set them as described.

### Describe the solution you'd like

The possibility to pass an icon for undetermined state like implemented for true and false in #3301 

### Describe alternatives you have considered

The only alternative we have is to find the correct element in the dom ourselves and change the icon. Now if you change anything in your html our code will break.

**Comment by Rakasch** [comment]
I had the same requirement.

I solved it by writing a IndifferentCheckbox component that wraps TriStateCheckbox and overrides the icon .pi-times:before with my own icon.
A Problem I encountered: I wanted to show "null" as highlighted (with a minus) and "false" as not highlighted and empty, as in your picture. And TriStateCheckbox has it the other way around.
So i don't have to overwrite all the styles, i instead swapped the values that I pass to TriStateCheckbox or are emited from TriStateCheckbox.
